atc/db/container_repository_test.go

```diff
@@ -1003,4 +1003,77 @@ var _ = Describe("ContainerRepository", func() {
 			})
 		})
 	})
+
+	Describe("DestroyUnknownContainers", func() {
+		var (
+			err                   error
+			workerReportedHandles []string
+			num                   int
+		)
+
+		BeforeEach(func() {
+			result, err := psql.Insert("containers").SetMap(map[string]interface{}{
+				"state":        atc.ContainerStateDestroying,
+				"handle":       "some-handle1",
+				"worker_name":  defaultWorker.Name(),
+				"hijacked":     false,
+				"discontinued": false,
+			}).RunWith(dbConn).Exec()
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(result.RowsAffected()).To(Equal(int64(1)))
+
+			result, err = psql.Insert("containers").SetMap(map[string]interface{}{
+				"state":        atc.ContainerStateCreated,
+				"handle":       "some-handle2",
+				"worker_name":  defaultWorker.Name(),
+				"hijacked":     false,
+				"discontinued": false,
+			}).RunWith(dbConn).Exec()
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(result.RowsAffected()).To(Equal(int64(1)))
+		})
+
+		JustBeforeEach(func() {
+			num, err = containerRepository.DestroyUnknownContainers(defaultWorker.Name(), workerReportedHandles)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		Context("when there are containers on the worker that are not in the db", func() {
```

Comment:
maybe this is captured somewhere, but, might be important to capture the case when the are no unknown handles in the db 😁

Reply:
thanks for the catch!